framework/TSA/AutoCorrelation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Auto Correlation Transformation for Characterization of Time-Series Data.
+"""
+import numpy as np
+
+from utils import InputData, InputTypes, xmlUtils
+from .TimeSeriesAnalyzer import TimeSeriesGenerator, TimeSeriesCharacterizer
+
+
+# utility methods
+class AutoCorrelation(TimeSeriesGenerator , TimeSeriesCharacterizer):
+  """
+    Perform AutoCorrelation Transformation on time-dependent data.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class cls.
+      @ Out, specs, InputData.ParameterInput, class to use for specifying input of cls.
+    """
+    specs = super(AutoCorrelation, cls).getInputSpecification()
+    specs.name = 'AutoCorrelation'
+    specs.description = """AutoCorrelation TimeSeriesAnalysis algorithm."""
+    specs.addSub(InputData.parameterInputFactory(
+      'nlags',
+      contentType=InputTypes.IntegerType,
+      descr=""" Number of lags to return autocorrelation for. """
+    ))
+    specs.addSub(InputData.parameterInputFactory(
+      'alpha',
+      contentType=InputTypes.FloatType,
+      descr=""" If a number is given, the confidence intervals for the given level are returned. For instance if alpha=.05, 
+      95 % confidence intervals are returned where the standard deviation is computed according to Bartlett”s formula. """
+    ))
+    return specs
+
+
+  def __init__(self, *args, **kwargs):
+    """
+      A constructor that will appropriately intialize a time-series analysis object
+      @ In, args, list, an arbitrary list of positional values
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    # general infrastructure
+    super().__init__(self, *args, **kwargs)
```

Comment:
do you need "self" in a `super()` call?

Reply:
I don't think you need self. I probably just missed removing it after I copied the file from wavelet.py.